app/models/ebean/Alert.java

```diff
@@ -0,0 +1,212 @@
+/**
+ * Copyright 2015 Groupon.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package models.ebean;
+
+import com.avaje.ebean.Model;
+import com.avaje.ebean.annotation.CreatedTimestamp;
+import com.avaje.ebean.annotation.UpdatedTimestamp;
+
+import java.sql.Timestamp;
+import java.util.UUID;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import javax.persistence.Version;
+
+/**
+ * Data model for alerts.
+ *
+ * @author Deepika Misra (deepika at groupon dot com)
+ */
+@Entity
+@Table(name = "alerts", schema = "portal")
+public class Alert extends Model {
+
+    @Id
+    @Column(name = "id")
+    private UUID _id;
+
+    @Version
+    @Column(name = "version")
+    private Long _version;
+
+    @CreatedTimestamp
+    @Column(name = "created_at")
+    private Timestamp _createdAt;
+
+    @UpdatedTimestamp
+    @Column(name = "updated_at")
+    private Timestamp _updatedAt;
+
+    @Column(name = "name")
+    private String _name;
+
+    @Column(name = "cluster")
+    private String _cluster;
+
+    @Column(name = "service")
+    private String _service;
+
+    @Column(name = "context")
+    private String _context;
+
+    @Column(name = "metric")
+    private String _metric;
+
+    @Column(name = "statistic")
+    private String _statistic;
+
+    @Column(name = "period")
+    private String _period;
+
+    @Column(name = "operator")
+    private String _operator;
+
+    @Column(name = "quantity_value")
+    private double _quantityValue;
+
+    @Column(name = "quantity_unit")
+    private String _quantityUnit;
+
+    @Column(name = "extensions")
+    private String _extensions;
```

Comment:
We had a good talk today about the purpose of metricsportal and how it interacts with the apps it's vending this data to.  Specifically, we talked about how we want metricsportal to be a rich UI to author this data, not a json datastore.  With this goal in mind, we should think about how we represent a rich set of information to vend to the clients.  I believe the best way to go about this is to break the extensions into actual models that represent the parts of the extensions map.  Right now we're mostly concerned with Nagios config.  I would like to see a table with nagios-specific config for each alert.  When we add additional configuration, we should add a table for those as well.  Mapping this into the current API would just be a matter of merging the additional tables in to the extensions map.  I would also like to see a v2 api that exposes a list of configuration objects along with their types, but this could be deferred. Does that make sense?